test.sh

```diff
@@ -1,51 +1,92 @@
+	if test -z "$testsuite" || test "test-$testsuite" = "$testname";then
+		$@ || failures=$( [ -n "$failures" ] && echo "$failures\\n$@" || echo "$@" )
+	fi
 }
 
-# ensure there is no trailing whitespace or other whitespace errors
-run-test git diff-tree --check $(git hash-object -t tree /dev/null) HEAD
+# only run test if it is explicitly selected, otherwise report it is skipped
+function skip-testsuite()
+{
+	testname=$(basename "$1" .sh)
+	# show skip message only when running full suite
+	if test -z "$testsuite";then
+		echo skipping "$@" "($REASON)"
+		blue SKIP
+	else
+		# if a skipped suite is explicity called, run it anyway
+		if test "test-$testsuite" == "$testname";then
+			run-testsuite "$@"
+		fi
+	fi
+}
 
-# ensure entries to authors file are sorted
-start=$(($(grep -n '^[[:space:]]*$' AUTHORS | awk -F ':' '{print $1}' | head -1) + 1))
-run-test diff <(tail -n +$start AUTHORS | sort) <(tail -n +$start AUTHORS)
+# used at the end to tell if everything went fine
+failures=''
 
-run-test ./test/test-gofmt.sh
-run-test ./test/test-yamlfmt.sh
-run-test ./test/test-bashfmt.sh
-run-test ./test/test-headerfmt.sh
-run-test ./test/test-commit-message.sh
-run-test ./test/test-govet.sh
-run-test ./test/test-examples.sh
-run-test ./test/test-gotest.sh
+run-testsuite ./test/test-misc.sh
+run-testsuite ./test/test-gofmt.sh
+run-testsuite ./test/test-yamlfmt.sh
+run-testsuite ./test/test-bashfmt.sh
+run-testsuite ./test/test-headerfmt.sh
+run-testsuite ./test/test-commit-message.sh
+run-testsuite ./test/test-govet.sh
+run-testsuite ./test/test-gotest.sh
+run-testsuite ./test/test-examples.sh
 
 # do these longer tests only when running on ci
 if env | grep -q -e '^TRAVIS=true$' -e '^JENKINS_URL=' -e '^BUILD_TAG=jenkins'; then
-	run-test ./test/test-shell.sh
-	#run-test ./test/test-gotest.sh --race	# XXX: temporarily disabled...
+	run-testsuite ./test/test-shell.sh
+	skip-testsuite ./test/test-gotest.sh --race	# XXX: temporarily disabled...
+else
+	REASON="CI server only test" skip-testsuite ./test/test-shell.sh
+	REASON="CI server only test" skip-testsuite ./test/test-gotest.sh --race	# XXX: temporarily disabled...
 fi
 
-run-test ./test/test-gometalinter.sh
+run-testsuite ./test/test-gometalinter.sh
+
 # FIXME: this now fails everywhere :(
-#run-test ./test/test-reproducible.sh
+skip-testsuite ./test/test-reproducible.sh
 
 # run omv tests on jenkins physical hosts only
 if env | grep -q -e '^JENKINS_URL=' -e '^BUILD_TAG=jenkins'; then
-	run-test ./test/test-omv.sh
+	run-testsuite ./test/test-omv.sh
+else
+	REASON="CI server only test" skip-testsuite ./test/test-omv.sh
 fi
-run-test ./test/test-golint.sh	# test last, because this test is somewhat arbitrary
+
+REASON="https://github.com/purpleidea/mgmt/issues/327" skip-testsuite ./test/test-crossbuild.sh
+
+run-testsuite ./test/test-golint.sh	# test last, because this test is somewhat arbitrary
 
 if [[ -n "$failures" ]]; then
-	echo 'FAIL'
+	echo
+	echo 'FAILURE'
```

Comment:
Please don't rename this line unless there's a good reason.

Reply:
The reason behind it is to distinct the conclusion of the entire run from the output of a single test, PASS/FAIL SUCCESS/FAILURE. Any other suggestion is welcome, if you don't like it I'm just as happy to keep it as FAIL.

Reply:
Let's keep this kind of churn to a minimum for now please. Sorry.